src/test/java/org/codehaus/plexus/archiver/zip/ArchiveFileComparator.java

```diff
@@ -84,119 +79,119 @@ private static Map getFileEntries( ZipFile file )
         } );
-        Assert.assertTrue( map2.isEmpty() );
+        assertTrue( map2.isEmpty() );
     }
 
-    public static void assertEquals( final ArchiveFile file1, final ZipFile file2, final String prefix )
-        throws Exception
+    public static void assertTarZipEquals( final org.codehaus.plexus.archiver.tar.TarFile cmp1, final ZipFile file2, final String prefix )
     {
         final Map<String, ZipArchiveEntry> map2 = getFileEntries( file2 );
-        forEachTarArchiveEntry( file1, new TarArchiveEntryConsumer()
-        {
-
-            public void accept( TarArchiveEntry ze1 )
-                throws IOException
-            {
-                final String name1 = ze1.getName();
-                final String name2 = prefix == null ? name1 : ( prefix + name1 );
-                ZipArchiveEntry ze2 = map2.remove( name2 );
-                Assert.assertNotNull( ze2 );
-                assertEquals( file1, ze1, file2, ze2 );
-            }
-
+        forEachTarArchiveEntry( cmp1, ze1 -> {
+            final String name1 = ze1.getName();
+            final String name2 = prefix == null ? name1 : ( prefix + name1 );
+            ZipArchiveEntry ze2 = map2.remove( name2 );
+            assertNotNull( ze2 );
+            assertTarZipEquals( cmp1, ze1, file2, ze2 );
         } );
-        Assert.assertTrue( map2.isEmpty() );
+        assertTrue( map2.isEmpty() );
     }
 
-    public static void assertEquals( org.apache.commons.compress.archivers.zip.ZipFile file1,
-                                     org.apache.commons.compress.archivers.zip.ZipFile file2, String prefix )
-        throws Exception
+    public static void assertZipEquals( ZipFile file1, ZipFile file2, String prefix )
     {
-        final Map map1 = getFileEntries( file1 );
-        final Map map2 = getFileEntries( file2 );
-        for ( Object o : map1.keySet() )
+        final Map<String, ZipArchiveEntry> map1 = getFileEntries( file1 );
+        final Map<String, ZipArchiveEntry> map2 = getFileEntries( file2 );
+        for ( String name1 : map1.keySet() )
         {
-            final String name1 = (String) o;
             final String name2 = prefix == null ? name1 : ( prefix + name1 );
-            ZipArchiveEntry ze1 = (ZipArchiveEntry) map1.get( name1 );
-            ZipArchiveEntry ze2 = (ZipArchiveEntry) map2.remove( name2 );
-            Assert.assertNotNull( "Missing " + name1 + "in file 2", ze2 );
-            assertEquals( file1, ze1, file2, ze2 );
+            ZipArchiveEntry ze1 = map1.get( name1 );
+            ZipArchiveEntry ze2 = map2.remove( name2 );
+            assertNotNull( "Missing " + name1 + "in file 2", ze2 );
+            assertZipEquals( file1, ze1, file2, ze2 );
         }
-        Assert.assertTrue( map2.isEmpty() );
+        assertTrue( map2.isEmpty() );
     }
 
+    private static List<TarArchiveEntry> tarEntries( TarFile file )
+    {
+        try
+        {
+            return Collections.list( file.getEntries() )
```

Comment:
I don't understand the question here, the getEntries() return an Enumeration, that's why to get the stream I need to convert it to a List first.

Reply:
What I meant is that `.collect(Collectors.toList())` returns `List` already. Why pass that to `Collectors.toList()` to convert it to `List`?

Reply:
Oh, right, sorry, anyway the new refactor doesn't do a `Collectors.toList()` anymore.

Essentially to be able to get a stream the easiest way is to convert the enumeration to a List first, (an Enumeration does not support Stream API OOTB).

Why a stream is needed? because is the easiest way to filter, transform, and collect the elements, the `TarFile.getEntries()` returns an `ArchiveEntries`, and to use the consumer a `TarArchiveEntry` is expected so a cast is needed (the current API is a bit weird).

In other words `file.getEntries()` returns an `Enumeration` and to convert an enumeration to a `List` I use the `Collections.list()`. (Collections != Collectors)